File: src/asp_net_backend/src/api/Models/Booking.cs
namespace Restify.API.Models;

public class Booking
{
	public uint				Id { get; set; }
	public uint				CustomerId { get; set; }
	public uint				LodgingId { get; set; }

	public IList<RoomBooking>	RoomBookings { get; }
	public Payment?				Payment { get; set; }
	public Person				Customer { get; }
	public Lodging				Lodging { get; }

	public Booking() {}
	
	public Booking(IList<RoomBooking> roomBookings)
	{
		RoomBookings = roomBookings;
	}
}

public enum BookingStatus
{
	Created,
	Confirmed,
	Cancelled,
	Finished
}

File: src/frontend/src/app/models/booking.ts
import { Lodging } from "./lodging";
import { Payment } from "./payment";
import { RoomBooking } from "./room_booking";

export class Booking
{
    constructor(
        public id:                  number,
        public lodgingId:           number,
        public customerId:          number,
        public roomBookings:        RoomBooking[],
        public payment:             Payment | null,
        public customerUserName:    string,
        public lodging:             Lodging | null
    ) { }
}

export enum BookingStatus
{
    Created   = 0,
    Confirmed = 1,
    Cancelled = 2,
    Finished  = 3 
}